### ISSUE TITLE
Connecting to GitHub with SSH (Bad owner or permissions on C:\\Users\\user/.ssh/config)

### ISSUE THREAD

**Issue by dnoren32** [susanBuck/dwa15-fall2018#37]
I know I'm late on this project but I'm trying to get it wrapped up and submitted asap. I'm  running windows 10 on my pc.

I tried connecting to GitHub with SSH and I keep getting the same error. I deleted the .ssh directory and did the steps over again several times and I keep getting the following when I go to test. 

```
C:\Users\alexa\.ssh
λ  ssh -T git@github.com
Bad owner or permissions on C:\\Users\\alexa/.ssh/config
```

I took to Google to troubleshoot and all I could figure out is how to check permissions of a folder (I think). This is what I tried.

```
C:\Users\alexa\.ssh
λ ls -l
total 10
-rw-r--r-- 1 alexa 197609   25 Sep 20 13:18 config
-rw-r--r-- 1 alexa 197609 1679 Sep 20 13:17 dwa
-rw-r--r-- 1 alexa 197609  403 Sep 20 13:17 dwa.pub
-rw-r--r-- 1 alexa 197609  803 Sep 20 15:52 known_hosts
```

I also have this same issue (same error) trying to go from GitHub to DigitalOcean. So I'm unable to connect the two. 

Thanks in advance for the help!!

**Comment by mdeforest** [comment]
Try looking at this thread #12 . Some other people had a similar issue.

**Comment by dnoren32** [comment]
@mdeforest Thanks for pointing that out. I looked through closed threads but must have just missed that one. Unfortunately, I just read through each post several times and tried the options that worked for everyone else but I'm still stuck with the same error :/

**Comment by mdeforest** [comment]
I am not on Windows so I can only help so much, but I would try bypassing the permission errors by specifying the key directly using `ssh -T -i %home%/.ssh/dwa git@github.com`. Maybe @susanBuck or @DoryAzar have some ideas? (Sorry for pulling you in Dory, but you're good at troubleshooting!)

**Comment by dnoren32** [comment]
@mdeforest Thanks for the tip! I tried that previously and just did it again and came up with the same problem.

```
C:\Users\alexa\.ssh
λ ssh -T -i %home%/.ssh/dwa git@github.com
Bad owner or permissions on C:\\Users\\alexa/.ssh/config
```

**Comment by DoryAzar** [comment]
Did you change the permissions to 600?
It needs to be:

https://user-images.githubusercontent.com/237711/45631223-73fb1500-ba68-11e8-859f-e317cca6f43f.png

**Comment by dnoren32** [comment]
@DoryAzar I believe so and it didn't work but I'm not convinced I necessarily did it right. I'm very new to this. Here is what it looked like.

```
C:\Users\alexa\.ssh
λ chmod 600 ~/.ssh/config

C:\Users\alexa\.ssh
λ ll
total 26
drwxr-xr-x 1 alexa    0 Sep 20 14:39 ./
drwxr-xr-x 1 alexa    0 Sep 20 13:13 ../
-rw-r--r-- 1 alexa   25 Sep 20 17:43 config
-rw-r--r-- 1 alexa 1679 Sep 20 13:17 dwa
-rw-r--r-- 1 alexa  403 Sep 20 13:17 dwa.pub
-rw-r--r-- 1 alexa  803 Sep 20 15:52 known_hosts

C:\Users\alexa\.ssh
λ  ssh -T git@github.com
Bad owner or permissions on C:\\Users\\alexa/.ssh/config
```

**Comment by susanBuck** [comment]
@DoryAzar I've noticed that too. The weird thing is it doesn't throw any errors or anything, it just doesn't have any impact.

**Comment by DoryAzar** [comment]
try `chmod -R 600 %home%/.ssh`

**Comment by DoryAzar** [comment]
there is also a potential windows slash vs “the rest of the world” slash problem...
`chmod -R 600 %home%\.ssh`

after you do this if you can `ll` again so that we see the output

**Comment by dnoren32** [comment]
@susanBuck  I got a permission denied error when I tried to create new keys. Should I have also deleted the dwa keys? I wouldn't think that would matter. 

```
C:\Users\alexa\.ssh
λ rm config

C:\Users\alexa\.ssh
λ ll
total 25
drwxr-xr-x 1 alexa    0 Sep 20 18:29 ./
drwxr-xr-x 1 alexa    0 Sep 20 13:13 ../
-rw-r--r-- 1 alexa 1679 Sep 20 13:17 dwa
-rw-r--r-- 1 alexa  403 Sep 20 13:17 dwa.pub
-rw-r--r-- 1 alexa  803 Sep 20 15:52 known_hosts

C:\Users\alexa\.ssh
λ ssh-keygen -t rsa -C "dan713@g.harvard.edu"
Generating public/private rsa key pair.
Enter file in which to save the key (C:\Users\alexa/.ssh/id_rsa):
Enter passphrase (empty for no passphrase):
Enter same passphrase again:
Saving key "C:\\Users\\alexa/.ssh/id_rsa" failed: Permission denied
```

@DoryAzar 
I ran into the same permission denied error.
 ```
C:\Users\alexa\.ssh
λ chmod -R 600 %home%/.ssh
chmod: changing permissions of 'C:\Users\alexa/.ssh': Permission denied
chmod: changing permissions of 'C:\Users\alexa/.ssh/known_hosts': Permission denied
```
Both of these instances are the first times I've gotten a ```Permission denied``` message.

**Comment by dnoren32** [comment]
@DoryAzar the result was the same regardless of the slash direction.

**Comment by DoryAzar** [comment]
That is good news. Did you change the keys on github after you regenerated?

**Comment by dnoren32** [comment]
@DoryAzar So I went through the process three times. Each time I deleted the key in github and added the newly generated one. Is that what you're confirming?

**Comment by DoryAzar** [comment]
@dnoren32 yes.
the permission denied is about not having the rights to change permissions.
so now try `elevate chmod -R 600 %home%\.ssh`

**Comment by dnoren32** [comment]
@DoryAzar Is this what we would expect to see?

```
C:\Users\alexa\.ssh
λ elevate chmod -R 600 %home%\.ssh

C:\Users\alexa\.ssh
λ ll
total 25
drwxr-xr-x 1 alexa    0 Sep 20 18:29 ./
drwxr-xr-x 1 alexa    0 Sep 20 13:13 ../
-rw-r--r-- 1 alexa 1679 Sep 20 13:17 dwa
-rw-r--r-- 1 alexa  403 Sep 20 13:17 dwa.pub
-rw-r--r-- 1 alexa  803 Sep 20 15:52 known_hosts
```

**Comment by susanBuck** [comment]
@dnoren32 If you'd like to troubleshoot this in real-time, feel free to message me in Slack. We can walk through each part step by step.

**Comment by dnoren32** [comment]
@susanBuck That would be awesome. Thank you!

**Comment by DoryAzar** [comment]
Did it work out for you @dnoren32?

**Comment by dnoren32** [comment]
@DoryAzar Yep, @susanBuck walked me through the full process again only we omitted the config folder and created the keys directly in the .ssh directory

**Comment by susanBuck** [comment]
Following-up for anyone else reading this thread - @dnoren32 and I debugged this in Slack and here's what the outcome was -

As noted above, when attempting to create a new key in his .ssh directory, he was getting a permissions error.

We could have tried and fixed the permissions, but we took the easy route and simply deleted the .ssh directory and recreated it. (I only suggested this because I could see the contents of his .ssh directory and knew there wasn't anything in there that wasn't easily recreatable. If you have other keys in your .ssh directory that you use outside this course, I would not suggest taking this route). 

Once he recreated his .ssh directory, he was able to generate a new ssh key using the default key name `id_rsa`. Because he was now using the default key name, he didn't need the SSH config file, and thus was not hitting the wall of the permission errors he initially created this post about.

From there, he deleted any existing keys he had saved in Github and DigitalOcean, then added his newly created `id_rsa.pub` key.

Upon doing this he tested his SSH connection to Github and it worked. 👍

When attempting to SSH into his Droplet, it failed, but this was expected because the Droplet had been created with an older key. 

To resolve this, I had him a) double check his new key was saved in his DigitalOcean settings and b) create a brand new Droplet, ensuring that he chose the option to add his new key during the creation process.

And that did the trick - he was able to SSH into his DigitalOcean Droplet.

So we got things working, but we did circumvent the initial problem about permissions on the SSH config file. If you're facing that issue and want more info - [see this reply I made to another issue](https://github.com/susanBuck/dwa15-fall2018/issues/12#issuecomment-423365249) as it includes video instructions I just made on how to correct the permissions.
